filebeat/input/udp/input.go

```diff
@@ -91,74 +118,22 @@ func NewInput(cfg *conf.C, outlet channel.Connector, context input.Context) (inp
 				},
 			}
 		}
-		_ = forwarder.Send(evt)
+
+		publisher.Publish(evt)
 
 		// This must be called after forwarder.Send to measure
 		// the processing time metric.
 		metrics.log(data, evt.Timestamp)
-	}
-
-	udp := udp.New(&config.Config, callback)
-
-	return &Input{
-		outlet:  out,
-		udp:     udp,
-		started: false,
-		metrics: metrics,
-	}, nil
-}
-
-func configID(config *conf.C) (string, error) {
-	var tmp struct {
-		ID string `config:"id"`
-	}
-	if err := config.Unpack(&tmp); err != nil {
-		return "", fmt.Errorf("error extracting ID: %w", err)
-	}
-	if tmp.ID != "" {
-		return tmp.ID, nil
-	}
-
-	var h map[string]interface{}
-	_ = config.Unpack(&h)
-	id, err := hashstructure.Hash(h, nil)
-	if err != nil {
-		return "", fmt.Errorf("can not compute ID from configuration: %w", err)
-	}
+	})
 
-	return fmt.Sprintf("%16X", id), nil
-}
+	log.Debugf("udp input '%v' initialized", ctx.ID)
```

Comment:
No need to include the ID in the message since the v2 framework includes the ID in the logger's context.

> {"log.level":"debug","@timestamp":"2022-12-06T22:05:07.433-0500","log.logger":"input.udp.udp","log.origin":{"file.name":"udp/input.go","file.line":129},"message":"udp input 'udp-udp.generic-eb7ec6cc-b99b-44b3-a26c-dc2f5bdc2451' initialized","service.name":"filebeat","id":"udp-udp.generic-eb7ec6cc-b99b-44b3-a26c-dc2f5bdc2451","host":"localhost:9514","ecs.version":"1.6.0"}